### DISCUSSION TITLE
Install my style, edit, and publish an updated version

### DISCUSSION THREAD

**Discussion by denilsonsa** [openstyles/stylus#1516] [Q&A]
I know I can write a userstyle from scratch and use the "Publish" button to publish it to userstyles.world. Afterwards, Stylus will keep this link between the local style and the remote one, so it's easy to publish updated versions.

I also know I can go to the [my available user styles](https://userstyles.world/user/denilsonsa) and download/install them in Stylus, and then they will be updated whenever I publish a new version there.

However, it seems those links are unidirectional:

* After publishing a style, I can go to the website and try to install it, and Stylus shows "Install" instead of "Reinstall".
* After downloading a style, Stylus doesn't provide me with a way to update it.

---

Some screenshots showing the issues…

Here's the style I created locally and then published through Stylus:
![image](https://user-images.githubusercontent.com/121676/209301706-2eb5acc1-dd54-4d30-b8d1-c5c8b13bcb44.png)

And when I open [the page for that user style](https://userstyles.world/style/7819/whatsapp-web-normal-scrollbars), I see:
![image](https://user-images.githubusercontent.com/121676/209301846-aed397b4-c5aa-4bb9-a220-5692f58af6c6.png)

If I click install, I get:
![image](https://user-images.githubusercontent.com/121676/209301963-a245cf34-7ab9-45e1-a0ab-a49ab5f2fe6c.png)

However, if I go to another style I have published (actually imported from the old website):
![image](https://user-images.githubusercontent.com/121676/209302207-fd13fd94-b26f-4e82-8858-2663801cd72e.png)
![image](https://user-images.githubusercontent.com/121676/209302274-13e68e25-ddfe-4304-8847-8f29213fa922.png)
![image](https://user-images.githubusercontent.com/121676/209302350-20b5aef2-c6f3-4f2a-b0f1-92ff65d1a0fd.png)

---

I wish downloading a style that I maintain would let me use Stylus to directly publish updates.
And I wish publishing updates would let Stylus to automatically check and download updates. (In case I update that style from another computer.)

Should this be moved to "Issues" instead of "Discussions"?

**Comment by tophf** [comment]
Without seeing the actual style I can only guess your original one isn't in [usercss mode](https://github.com/openstyles/stylus/wiki/Writing-UserCSS) so it doesn't have `@namespace`, which is why it's not recognized as the same when installing.

**Reply by denilsonsa** [reply]
I've linked the style in the middle of the post, but indeed it's easy to miss that link. Here: https://userstyles.world/style/7819/whatsapp-web-normal-scrollbars

I even compared, both the local copy and the remote one are identical.

And, yes, it includes `@namespace`, using the default namespace that Stylus suggested me.

---

Still, that's only half of the problem, the other half is being able to publish updates directly from Stylus to my downloaded styles. (But to be honest I haven't tried that. I only noticed the "publish" UI of the downloaded styles looks different.)

---

I forgot to mention: using Stylus on Firefox.

**Reply by tophf** [reply]
If I followed the description correctly, it works for me. Try the current [dev version](https://github.com/openstyles/stylus/archive/refs/heads/master.zip) (unpack and install it in `about:debugging` after you make a backup of the styles). To undo installation you'll have to restart Firefox.

**Comment by denilsonsa** [comment]
Months later…

While the UI could be a bit more intuitive sometimes, part of my trouble was learning how the UI works for the first (few) time(s). I'm closing this discussion, as I don't think I have anything to contribute for now. And thank you for taking the time to reply to me.
